src/web-auth/transaction-manager.js

```diff
@@ -11,20 +11,21 @@ function TransactionManager(options) {
 
 TransactionManager.prototype.process = function(options) {
   var transaction;
+  var responseType = options.responseType || '';
 
-  if (options.responseType.indexOf('code') !== -1) {
+  if (responseType.indexOf('code') !== -1) {
     return options;
   }
 
-  if (options.responseType.indexOf('id_token') !== -1 && !!options.nonce) {
+  if (responseType.indexOf('id_token') !== -1 && !!options.nonce) {
     return options;
   }
 
   transaction = this.generateTransaction(options.appState, options.state, options.nonce);
 
   options.state = transaction.state;
 
-  if (options.responseType.indexOf('id_token') !== -1) {
+  if (responseType.indexOf('id_token') !== -1) {
     options.nonce = transaction.nonce;
```

Comment:
What happens if the user has defined his own options.nonce?

Reply:
it will be already out of this codepath because of https://github.com/auth0/auth0.js/pull/528#discussion-diff-146922975R20